client/app/routes/market.tsx

```diff
@@ -49,18 +75,43 @@ export default function Market() {
 	const { categories, products, sellerProfile } =
 		useLoaderData<typeof loader>();
 	const { user } = useRouteLoaderData<typeof rootLoader>("root") || {};
+	const navigate = useNavigate();
+
+	const [searchQuery, setSearchQuery] = useState<string>("");
+	const [selectedCategory, setSelectedCategory] = useState<string>("all");
+
+	const handleSearchInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		const value = e.target.value;
+		setSearchQuery(value);
+		navigate(
+			`/market?q=${encodeURIComponent(value)}&category=${encodeURIComponent(selectedCategory)}`, // Encode category name
+		);
```

Comment:
Move this to an effect (ie `useEffect`)

And also when you refresh the page and the params are in the URL, the search field and the category should reflect those values.